### PR TITLE
fix: Allow using custom header for principal propagation

### PR DESCRIPTION
**Pull request by marikaner**
Related to this issue: https://answers.sap.com/questions/13416513/sap-cloud-sdk-core-giving-issues-with-principal-pr.html.

The user tried to set the pp header per custom headers. This did not work, because it isn't treated as auth header. Although this header was set, the error message stated, that the header was missing. This PR improves this behavior.

### CODE REVIEW

File: packages/core/src/connectivity/scp-cf/authorization-header.spec.ts [hunk 1]
```diff
@@ -59,28 +81,21 @@ describe('buildAuthorizationHeaders', () => {
...
+      const headers = await buildAuthorizationHeaders(
+        principalPropagationDestination
+      );
       checkHeaders(headers);
+    });
 
-      delete destination!.proxyConfiguration!.headers![
-        'SAP-Connectivity-Authentication'
-      ];
+    it('does not throw on Principal Propagation', async () => {
```
Comment (FrankEssenberger) [inline packages/core/src/connectivity/scp-cf/authorization-header.spec.ts:91]: it says does not throw but in the test it is expected to throw -> confused.

File: packages/core/src/connectivity/scp-cf/authorization-header.ts [hunk 2]
```diff
@@ -61,11 +61,21 @@ function hasAuthHeaders(destination: Destination): boolean {
...
+}
+
 export async function getAuthHeaders(
   destination: Destination,
   customHeaders?: Record<string, any>
 ): Promise<Record<string, string>> {
-  const customAuthHeaders = pickIgnoreCase(customHeaders, 'authorization');
+  const customAuthHeaders = getCustomAuthHeaders(destination, customHeaders);
 
   if (Object.keys(customAuthHeaders).length && hasAuthHeaders(destination)) {
```
Comment (FrankEssenberger) [inline packages/core/src/connectivity/scp-cf/authorization-header.ts:80]: the `hasAuthHeaders` is not triggered in the `SAP-Connectivity-Authentication` case right? I would also like to have a warning is this is done.

**Review by FrankEssenberger — CHANGES_REQUESTED** [review]
LGTM but there are to git checks and I have two minor comments.

File: packages/core/src/connectivity/scp-cf/authorization-header.spec.ts (hunk 1 above)
Comment (marikaner) [reply packages/core/src/connectivity/scp-cf/authorization-header.spec.ts:91]: Fixed.

File: packages/core/src/connectivity/scp-cf/authorization-header.ts (hunk 2 above)
Comment (marikaner) [reply packages/core/src/connectivity/scp-cf/authorization-header.ts:80]: It does. It just checks for authentication type.

Comment (FrankEssenberger) [reply packages/core/src/connectivity/scp-cf/authorization-header.ts:80]: And we do not want to add a log statement if the `SAP-Connectivity-Authentication` is set via the custom header and is overwriting the automatic one?

Comment (marikaner) [reply packages/core/src/connectivity/scp-cf/authorization-header.ts:80]: I meant, that the warning is logged in case there is a custom header.

Comment (marikaner) [reply packages/core/src/connectivity/scp-cf/authorization-header.ts:80]: But I'll add a test for this.

**Review by FrankEssenberger — APPROVED** [review]
LGTM - as discussed small renaming of `hasAuthHeader`

**Pull request merged**